File: scala/chain.scala
package com.example.chain

// T1 transitive chain (flat scala — 3 package-level objects).
object Origin { val ORIGIN_VALUE: String = "T1.origin" }        // T1.origin.def

object Middle { val MIDDLE_VALUE: String = Origin.ORIGIN_VALUE } // T1.middle.reexport

object Deep   { val VALUE_ALIAS: String = Middle.MIDDLE_VALUE }  // T1.deep.reexport

File: elixir/chain.ex
defmodule Chain.Origin do
  @origin_value "T1.origin"                              # T1.origin.def
  def value, do: @origin_value
end

defmodule Chain.Middle do
  # Re-export — Elixir has no symbol re-export; we delegate.
  defdelegate value, to: Chain.Origin                    # T1.middle.reexport
end

defmodule Chain.Deep do
  defdelegate value_alias, to: Chain.Middle, as: :value  # T1.deep.reexport
end

defmodule Chain do
  def run do
    IO.puts("transitive: #{Chain.Deep.value_alias()}")
  end
end
